client/reader/get-helpers.ts

```diff
@@ -92,8 +92,8 @@ export const getFeedUrl = ( { feed, site, post }: GetFeedUrlArgs = {} ) => {
  * Given a feed, site, or post: return the site icon. return false if one could not be found.
  */
 export function getPostIcon( post: Partial< ReaderPost > ): string | undefined {
-	if ( typeof post?.site_icon === 'object' ) {
-		return post.site_icon?.img;
+	if ( post?.site_icon && typeof post?.site_icon === 'object' && 'img' in post.site_icon ) {
```

Comment:
Can we replace this with `post?.site_icon?.img`?

Reply:
You're right. I think we can simplify this to `if ( typeof post?.site_icon === 'object' && post?.site_icon?.img )` I think we still need `typeof post?.site_icon === 'object'` because I get a TypeScript lint error without it.